lib/Heap_List.thy

```diff
@@ -267,4 +267,73 @@ lemma heap_ls_next_not_in:
   \<Longrightarrow> np \<notin> set xs"
   by (fastforce dest!: heap_ls_prev_cases[simplified])
 
+(* more on heap_path *)
+
+(* moved from ListLibLemmas *)
+lemma distinct_inj_middle: "distinct list \<Longrightarrow> list = (xa @ x # xb) \<Longrightarrow> list = (ya @ x # yb) \<Longrightarrow> xa = ya \<and> xb = yb"
```

Comment:
nit: while we're moving it, rename all variables?

Reply:
How they are currently named makes enough sense to me, but what change are you suggesting?

Reply:
Maybe `prex @ x # postx` and `prey @ y # posty`? Its just that `xa` and `xb` look like auto-generated names. But this is a nitpick.